cmd/commands/cluster.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2022 The Codefresh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/config"
+	"github.com/codefresh-io/cli-v2/pkg/log"
+	"github.com/codefresh-io/cli-v2/pkg/store"
+	"github.com/codefresh-io/cli-v2/pkg/util"
+	cdutil "github.com/codefresh-io/cli-v2/pkg/util/cd"
+	kustutil "github.com/codefresh-io/cli-v2/pkg/util/kust"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/argoproj-labs/argocd-autopilot/pkg/kube"
+	"github.com/juju/ansiterm"
+	"github.com/spf13/cobra"
+	kusttypes "sigs.k8s.io/kustomize/api/types"
+)
+
+type (
+	ClusterAddOptions struct {
+		runtimeName string
+		kubeContext string
+		dryRun      bool
+		kubeFactory kube.Factory
+	}
+)
+
+var minimalSupportedVersion = semver.MustParse("99.99.99") // to be updated after app-proxy is ready
+
+func NewClusterCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Hidden:            true, // until app-proxy is working correctly
+		Use:               "cluster",
+		Short:             "Manage clusters of Codefresh runtimes",
+		PersistentPreRunE: cfConfig.RequireAuthentication,
+		Args:              cobra.NoArgs, // Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+			exit(1)
+		},
+	}
+
+	cmd.AddCommand(NewClusterAddCommand())
+	cmd.AddCommand(NewClusterListCommand())
+
+	return cmd
+}
+
+func NewClusterAddCommand() *cobra.Command {
+	var (
+		opts ClusterAddOptions
+		err  error
+	)
+
+	cmd := &cobra.Command{
+		Use:     "add RUNTIME_NAME --context CONTEXT_NAME",
```

Comment:
should not have flags in the `USE` string, only positional params.
This should be an example.